marshal.go

```diff
@@ -983,6 +1000,23 @@ func (d *Decoder) unwrapPointer(mtype reflect.Type, tval interface{}, mval1 *ref
 	return mval, nil
 }
 
+func (d *Decoder) unmarshalText(tval interface{}, mval reflect.Value) error {
+	var s string
```

Comment:
I think you could replace this with
```suggestion
	var buf bytes.Buffer
	fmt.Fprint(&buf, tval)
	return callTextUnmarshaler(mval, buf.Bytes())
```